src/modinfo.cpp

```diff
@@ -312,6 +312,22 @@ bool ModInfo::checkAllForUpdate(PluginContainer *pluginContainer, QObject *recei
     }
   }
 
+  // Detect invalid source games
+  for (auto itr = games.begin(); itr != games.end(); ++itr) {
+    auto gamePlugins = pluginContainer->plugins<IPluginGame>();
+    IPluginGame* gamePlugin = qApp->property("managed_game").value<IPluginGame*>();
+    for (auto plugin : gamePlugins) {
+      if (plugin != nullptr && plugin->gameShortName().compare(*itr, Qt::CaseInsensitive) == 0) {
+        gamePlugin = plugin;
+        break;
+      }
+    }
+    if (gamePlugin != nullptr && gamePlugin->gameNexusName().isEmpty()) {
+      log::warn("{}", tr("The update check has found a mod with a Nexus ID and source game of %1, but this game is not a valid Nexus source.").arg(gamePlugin->gameName()));
+      itr = games.erase(itr);
```

Comment:
This looks wrong to me because of `++itr` in the loop. It'll either skip the next element or be undefined behaviour for the last element because it's `++end()`, which will probably be an infinite loop.